app/models/vendor_code.rb

```diff
@@ -7,11 +7,8 @@ class VendorCode < ActiveRecord::Base
                    length: { maximum: 255, message: "longer than 255 characters" }
 	
 	
-  def assign_to(current_user)
-    self.update_attributes(user: current_user, user_name: current_user.name, email: current_user.email)
-    vendor = self.vendor
-    vendor.update_attribute(:usedCodes, vendor.usedCodes + 1)
-    vendor.update_attribute(:unclaimCodes, vendor.unclaimCodes - 1)
+  def associate_with(user)
+    self.update(user_id: user.id, user_name: user.name, email: user.email)
   end  
```

Comment:
Alex,
I don't see why should the number of used and unclaimed vendor codes remain unchanged for the first user login even though that user switched to the vendor account. 
Contrary, consider the case of 50 uploaded vendor codes and 60 redeemify code loggers (users). Some 10 of them will be alerted with 'Some offers are not available at this time, please come back later' which would seem right. However, the vendor home page will show 0 unclaimed codes and 50 unused codes on the bar chart thus misleading the vendor of no codes actually used an no need for additional uploads.